feat: add extra hosts option

Haegi opened the pull request:
### Context

At the moment gnomock doesn't support adding extra hosts to the container.
This PR adds a `WithExtraHosts` option, which implements exactly that.
If the option is provided, the extra hosts will be added to the container and otherwise it will just receive an empty array and it will do nothing with it. 

### Notes

I am not sure on a proper test for it.
Maybe I could run some busybox and check if the specified host is reachable.

orlangure commented:
Hi @Haegi and thank you for the contribution! I wasn't aware of this flag, this is really cool!
I like your suggested test scenario. Can you update the PR with it?
Thanks!😻

Haegi commented:
> Hi @Haegi and thank you for the contribution! I wasn't aware of this flag, this is really cool! I like your suggested test scenario. Can you update the PR with it? Thanks!😻

I just added some tests and I also adapted the PR description.
However somehow the k3s tests fail (also locally without the change in my case). Do you have any idea about it or should I have a deeper look at it?

orlangure commented:
I'll take a look, I guess something broke in `k3s` preset😿

Haegi commented:
> I'll take a look, I guess something broke in `k3s` preset😿

Thanks :)

orlangure commented:
Hey, I fixed the k3s preset issue in #760, please rebase from `master`, the build should be OK😼

Haegi commented:
> Hey, I fixed the k3s preset issue in #760, please rebase from `master`, the build should be OK😼

Thanks @orlangure! I rebased from master but it seems like the github action is now broken. Is this because of my change and should I have a look at it?

orlangure commented:
@Haegi, no this is just a flaky job that I don't have capacity to look into. I will merge and prepare a release this weekend, everything looks great. Thank you!
